src/tcutility/analysis/report/formatters/xyz.py

```diff
@@ -0,0 +1,122 @@
+
+import scm.plams as plams
+from tcutility import results
+from tcutility.results import Result
+
+
+@dataclass
+class XYZData:
+    name: Union[str, None] = None
+    E: Union[float, None] = None
+    H: Union[float, None] = None
+    G: Union[float, None] = None
+    num_imag_modes: Union[int, None] = None
+    imag_freqs: Union[List[float], None] = None
+    molecule: Union[plams.Molecule, None] = None
+
+
+T = TypeVar("T")
+
+
+def try_to_get_property(obj: Result, prop: str, def_type: T) -> Union[T, None]:
+    try:
+        property_ = getattr(obj.properties.energy, prop)  # type: ignore  # Result object has no static typing
+        if property_ is not None and property_:
+            return property_
+    except AttributeError:
+        return None
+
+
+def check_if_job_is_suitable_for_xyz_format(obj: Union[Result, str]) -> bool:
+    """
+    Only jobs that are geometry optimizations, single points, or transition state searches are suitable for XYZ format.
+    Other calculations such as PES scans, IRCs, etc. are not suitable.
+    """
+    if isinstance(obj, str):
+        obj = results.read(obj)
+
+    if obj is None or not obj:
+        return False
+
+    try:
+        task = obj.input.task  # type: ignore  # Result object has no static typing
+    except AttributeError:
+        task = "unknown"
+
+    if not task:
+        return False
+
+    return task.lower() in ["geometryoptimization", "singlepoint", "transitionstatesearch"]  # type: ignore  # Result object has no static typing
+
+
+def get_data_for_xyz_format(obj: results.Result) -> XYZData:
+    return_data = XYZData(E=None, H=None, G=None, num_imag_modes=None, imag_freqs=None, molecule=None)
+    if not obj.properties or obj.properties is None:
+        return return_data
+
+    if isinstance(obj.files.root, str):  # type: ignore  # obj is not None
+        return_data.name = pl.Path(obj.files.root).stem  # type: ignore  # obj is not None
+
+    return_data.E = try_to_get_property(obj, "bond", 0.0)  # Total electronic energy
+    return_data.H = try_to_get_property(obj, "enthalpy", 0.0)  # Enthalpy
+    return_data.G = try_to_get_property(obj, "gibbs", 0.0)  # Gibbs free energy
+
+    # add imaginary frequency if we have one
+    if obj.properties.vibrations:  # type: ignore  # Result object has no static typing
+        if obj.properties.vibrations is not None:
+            return_data.num_imag_modes = obj.properties.vibrations.number_of_imag_modes  # type: ignore  # Result object has no static typing
+            return_data.imag_freqs = [f for f in obj.properties.vibrations.frequencies if f < 0]  # type: ignore  # Result object has no static typing
+
+    return_data.molecule = obj.molecule.output  # type: ignore  # Result object has no static typing
+    return return_data
+
+
+def add_title(title: str) -> str:
+    # Add title to the document
+    return f"<b>{title}</b><br>"
+
+
+def format_xyz(data: XYZData, title: Union[str, None]) -> str:
```

Comment:
Should this be a separate function or part of the XYZFormatter class?

Reply:
I prefer to have it as function because 
1) the function is more easy to test (although there are currently no tests for it)
2) an external user may want to design its own XYZFormatter which has as only requirement the format() method in class (WordFormatter is the interface class with as only goal to communicate to the programmer which method and arguments must be present and used). Adding other methods would, in my opinion, confuse the programmer because you add extra methods to the class that are not part of its main function: format()

Reply:
Clear!